src/index/vchordrq/am/am_build.rs

```diff
@@ -1327,9 +1340,16 @@ fn make_internal_build(
         samples
     };
     let mut result = Vec::<Structure<Normalized>>::new();
+    let approximation = match internal_build.kmeans_algorithm {
+        KMeansAlgorithm::Lloyd {} => false,
+        KMeansAlgorithm::Approx {} => true,
+    };
+    if !approximation && reduction {
+        pgrx::warning!("using dimension reduction without approximation could be much slower.");
```

Comment:
I guess that it means here is that the acceleration is not maximized. I'm not sure whether it's useful for us to set this warning. Maybe, some people use dimension reduction simply to reduce memory usage.